base_layer/common_types/src/emoji.rs

```diff
@@ -172,37 +164,45 @@ impl Display for EmojiId {
 
 // TODO: We have to add more details
 #[derive(Debug, Error)]
-#[error("emoji id error")]
-pub struct EmojiIdError;
+pub enum EmojiIdError {
+    #[error("Byte array error: {0}")]
+    ByteArray(#[from] tari_utilities::ByteArrayError),
+    #[error("Hex error in EmojiId: {0}")]
+    HexError(#[from] tari_crypto::tari_utilities::hex::HexError),
+    #[error("Unsupported char for EmojiId: '{0}'")]
+    UnsupportedChar(char),
+    #[error("Invalid checksum of EmojiId")]
+    InvalidChecksum,
+}
 
 #[cfg(test)]
 mod test {
+    use anyhow::Error;
     use tari_crypto::tari_utilities::hex::Hex;
 
     use crate::{emoji::EmojiId, types::PublicKey};
 
     #[test]
-    fn convert_key() {
-        let pubkey = PublicKey::from_hex("70350e09c474809209824c6e6888707b7dd09959aa227343b5106382b856f73a").unwrap();
-        let eid = EmojiId::from_hex("70350e09c474809209824c6e6888707b7dd09959aa227343b5106382b856f73a").unwrap();
+    fn convert_key() -> Result<(), Error> {
```

Comment:
No problem with this, but let's not change this everywhere. From experience, the unwraps in tests are more useful than returning a result from tests because they give you the line of the unwrap/panic whereas returning a result just gives you the failure - in longer tests, it becomes difficult to hunt for the line that failed.